WordPress/Classes/ViewRelated/Reader/Tab Navigation/ReaderTabView.swift

```diff
@@ -188,3 +205,41 @@ private extension ReaderTabView {
         static let dividerColor: UIColor = .divider
     }
 }
+
+// MARK: - ReaderNavigationMenuDelegate
+
+extension ReaderTabView: ReaderNavigationMenuDelegate {
+
+    func scrollViewDidScroll(_ scrollView: UIScrollView, velocity: CGPoint) {
+        let isContentOffsetNearTop = scrollView.contentOffset.y < scrollView.frame.height / 2
+        let isUserScrollingDown = velocity.y < -400
+        let isUserScrollingUp = velocity.y > 400
+
+        if !isMenuHidden && !isContentOffsetNearTop && isUserScrollingDown {
+            updateMenuDisplay(hidden: true)
+        }
+
+        if isMenuHidden && isUserScrollingUp {
+            updateMenuDisplay(hidden: false)
+        }
+
+        // Accounts for a user scrolling slowly enough to not trigger displaying the menu near the top of the content
+        if isMenuHidden && isContentOffsetNearTop && velocity.y > 0 {
+            updateMenuDisplay(hidden: false)
+        }
+
+    }
+
+    func scrollViewWillEndDragging(_ scrollView: UIScrollView, withVelocity velocity: CGPoint, targetContentOffset: UnsafeMutablePointer<CGPoint>) {
+        let isTargetContentNearTop = targetContentOffset.pointee.y < scrollView.frame.height / 2
+
+        // Accounts for the case where a user quickly swipes the scroll view without holding their
+        // finger on it.
+        // Note: velocity here is opposite of the velocity in `scrollViewDidScroll`. Postive values
```

Comment:
Nit: Just a tiny typo 😄 

```suggestion
        // Note: velocity here is opposite of the velocity in `scrollViewDidScroll`. Positive values
```

Reply:
Fixed in https://github.com/wordpress-mobile/WordPress-iOS/pull/22353/commits/71b832cd25189d1772f1a0ff37e7982a4bcf764f, thanks!